modules/meta/arrow-meta-prototype/testing-plugin/src/test/kotlin/arrow/meta/plugin/testing/CompilationTest.kt

```diff
@@ -0,0 +1,35 @@
+package arrow.meta.plugin.testing
+
+import org.junit.Test
+
+class CompilationTest {
+
+  @Test
+  fun `metadebug consideration works as expected`() {
+
+    testCompilation(
+      CompilationData(
+        sourceFileName = "Example.kt",
+        sourceContent = contentFromResource(javaClass, "Example.kt.source"),
+        generatedFileContent = contentFromResource(javaClass, "Example.kt.meta"),
+        generatedClasses = arrayListOf("ExampleKt", "ForId2", "Id2", "ForId2\$Companion"),
```

Comment:
😍

Reply:
I love you like it! I tried to wrap the external library and to hide all the complexity.